nemo_rl/models/policy/megatron_policy_worker.py

```diff
@@ -291,8 +291,8 @@ def __init__(
         model_cfg.context_parallel_size = self.cfg["megatron_cfg"]["context_parallel_size"] # not supported right now
         model_cfg.bf16 = self.dtype == torch.bfloat16
         model_cfg.fp16 = self.dtype == torch.float16
-        model_cfg.params_dtype = torch.float32  # amp
-        model_cfg.pipeline_dtype = torch.float32  # dtype_map[self.cfg["pipeline_dtype"]]
+        model_cfg.params_dtype = self.dtype  # amp
```

Comment:
Wouldn't this disable AMP? It'd be master weights in low precision. We can make this configurable if you need a lower precision master weight for DSv2